src/primitives/transaction.h

```diff
@@ -340,4 +344,34 @@ class CTransaction
 
 };
 
+/** A mutable version of the Bitcoin CTransaction. This is only used to adapt the Bitcoin transactions in the test suite
+ * to the Gridcoin format.
+ */
+struct CMutableBitcoinTransaction
+{
+    int32_t nVersion;
+    std::vector<CTxIn> vin;
+    std::vector<CTxOut> vout;
+    uint32_t nLockTime;
+
+    CMutableBitcoinTransaction();
+
+    CMutableBitcoinTransaction(const CTransaction& tx);
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(nVersion);
+        READWRITE(vin);
+        READWRITE(vout);
+        READWRITE(nLockTime);
+    }
+
+    uint256 GetHash() const
+    {
+        return SerializeHash(*this);
+    }
+};
+
```

Comment:
Should be in the test file in my opinion.

Reply:
I tried that, but the inverse constructor in CTransaction using CMutableBitcoinTransaction as an argument didn't like it.

Reply:
Can't you do a normal function to do the conversion instead of a constructor?

Reply:
I suppose so...